### PR TITLE
Fixed compiling bug when CPU_ONLY=ON

### PR DESCRIPTION
**Pull request by listenlink**
Hi @gongzg 

This patch fix compiling bug when set CPU_ONLY=ON

### CODE REVIEW

File: src/caffe/layers/smooth_L1_loss_layer.cpp [hunk 1]
```diff
@@ -98,10 +98,6 @@ void SmoothL1LossLayer<Dtype>::Backward_cpu(const vector<Blob<Dtype>*>& top,
   }
 }
 
-#ifdef CPU_ONLY
-STUB_GPU(SmoothL1LossLayer);
```
Comment (gongzg) [inline src/caffe/layers/smooth_L1_loss_layer.cpp]: Why remove this?

Comment (listenlink) [reply src/caffe/layers/smooth_L1_loss_layer.cpp]: at the header file, the gpu function using 
```
#ifdef USE_CUDA
  virtual void Forward_gpu(const vector<Blob<Dtype>*>& bottom,
      const vector<Blob<Dtype>*>& top);
#endif
```
so, in the cpu_only mode, there is no declaration of the gpu_function, add the STUB_GPU will bring can't find the decaration bug.

**Comment by gongzg** [comment]
@listenlink Then it's better to modify the header file to keep align with other layers.

**Comment by listenlink** [comment]
@gongzg have modify to common method to align with other layers.

**Pull request merged**